### PR TITLE
Use Testify.

### PR DESCRIPTION
**Pull request by ldez**
Use Testify in lib tests

### CODE REVIEW

File: acme/client_test.go [hunk 1]
```diff
@@ -169,29 +154,55 @@ func TestValidate(t *testing.T) {
...
-		{"GET-valid", []string{"pending", "valid"}, ""},
-		{"GET-invalid", []string{"pending", "invalid"}, "Error"},
+		{
+			name:     "POST-unexpected",
+			statuses: []string{"weird"},
+			want:     "unexpected",
+		},
+		{
+			name:     "POST-valid",
+			statuses: []string{"valid"}},
```
Comment (wyattjoh) [inline acme/client_test.go]: Fix weird bracket

File: acme/client_test.go [hunk 2]
```diff
@@ -169,29 +154,55 @@ func TestValidate(t *testing.T) {
...
+			want:     "Error",
+		},
+		{
+			name:     "GET-unexpected",
+			statuses: []string{"pending", "weird"},
+			want:     "unexpected",
+		},
+		{
+			name:     "GET-valid",
+			statuses: []string{"pending", "valid"}},
```
Comment (wyattjoh) [inline acme/client_test.go]: Another bracket

File: acme/client_test.go [hunk 3]
```diff
@@ -301,7 +305,7 @@ func writeJSONResponse(w http.ResponseWriter, body interface{}) {
 }
 
 // stubValidate is like validate, except it does nothing.
-func stubValidate(j *jws, domain, uri string, chlng challenge) error {
+func stubValidate(_ *jws, _, _ string, _ challenge) error {
```
Comment (wyattjoh) [inline acme/client_test.go]: tbh, the variables before I felt were more readable. This one is really hard to understand.

Comment (ldez) [reply acme/client_test.go]: the parameters names are define in `type validateFunc func(j *jws, domain, uri string, chlng challenge) error` 

https://github.com/xenolf/lego/blob/master/acme/client.go#L55

**Pull request merged**